toolkit/tools/imagegen/attendedinstaller/primitives/enumfield/enumfield.go

```diff
@@ -0,0 +1,252 @@
+	labelWidth int
+
+	// Options to choose from
+	options []string
+
+	// Index of the currently selected option
+	selectedOption int
+
+	// The label color.
+	labelColor tcell.Color
+
+	// The label color when this item is selected.
+	labelColorActivated tcell.Color
+
+	// The text color.
+	textColor tcell.Color
+
+	// The text color when this item is selected.
+	textColorActivated tcell.Color
+
+	// The background color.
+	backgroundColor tcell.Color
+
+	// The background color when this item is selected.
+	backgroundColorActivated tcell.Color
+
+	// A callback invoked when the user leaves this form item
+	onFinished func(tcell.Key)
+
+	// A callback invoked when this primitive receives focus.
+	onFocus func()
+
+	// A callback invoked when this primitive loses focus.
+	onBlur func()
+}
+
+// NewEnumField returns a new navigation bar.
+func NewEnumField(options []string) *EnumField {
+	return &EnumField{
+		Box:                      tview.NewBox(),
+		backgroundColor:          tview.Styles.PrimitiveBackgroundColor,
+		backgroundColorActivated: tview.Styles.ContrastBackgroundColor,
+		labelColor:               tview.Styles.SecondaryTextColor,
+		labelColorActivated:      tview.Styles.ContrastSecondaryTextColor,
+		textColor:                tview.Styles.PrimaryTextColor,
+		textColorActivated:       tview.Styles.PrimaryTextColor,
+		options:                  options,
+	}
+}
+
+// SetLabelColor sets the color of button text.
+func (n *EnumField) SetLabelColor(color tcell.Color) *EnumField {
+	n.labelColor = color
+	return n
+}
+
+// SetLabelColorActivated sets the color of button text when the button is
+// selected.
+func (n *EnumField) SetLabelColorActivated(color tcell.Color) *EnumField {
+	n.labelColorActivated = color
+	return n
+}
+
+// SetNavBackgroundColor sets the background color.
+func (n *EnumField) SetFieldBackgroundColor(color tcell.Color) *EnumField {
+	n.backgroundColor = color
+	return n
+}
+
+// SetBackgroundColorActivated sets the background color of the button text when
+// the button is selected.
+func (n *EnumField) SetBackgroundColorActivated(color tcell.Color) *EnumField {
+	n.backgroundColorActivated = color
+	return n
+}
+
+// Draw renders this primitive onto the screen.
+func (n *EnumField) Draw(screen tcell.Screen) {
+
+	// Update dimensions
+	n.Box.Draw(screen)
+
+	// Obtain dimensions
+	x, y, width, height := n.GetInnerRect()
+	rightLimit := x + width
+	if height < 1 || rightLimit <= x {
+		return
+	}
+
+	// Draw label.
+	if n.labelWidth > 0 {
+		labelWidth := n.labelWidth
+		if labelWidth > rightLimit-x {
+			labelWidth = rightLimit - x
+		}
+		tview.Print(screen, n.label, x, y, labelWidth, tview.AlignLeft, n.labelColor)
+		x += labelWidth
+	} else {
+		_, drawnWidth := tview.Print(screen, n.label, x, y, rightLimit-x, tview.AlignLeft, n.labelColor)
+		x += drawnWidth
+	}
+
+	// get as much space as needed or available
+	fieldWidth := math.MaxInt32
+	if rightLimit-x < fieldWidth {
```

Comment:
Is it even possible for this to not be true, since we're comparing against `math.MaxInt32`? Can `x` be negative for instance?

Reply:
rightLimit and x are `int`s, which can be `int64`. I'm not sure if that's likely to happen; this is a modified version of one of the primitives from the tview.